doc/read-the-docs-site/tutorials/AuctionValidator.hs

```diff
@@ -0,0 +1,194 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+module AuctionValidator where
+
+import PlutusCore.Default qualified as PLC
+import PlutusLedgerApi.V1 (POSIXTime, PubKeyHash, Value, adaSymbol, adaToken, singleton)
+import PlutusLedgerApi.V1.Address (pubKeyHashAddress)
+import PlutusLedgerApi.V1.Interval (contains)
+import PlutusLedgerApi.V2 (Datum (..), OutputDatum (..), ScriptContext (..), TxInfo (..), TxOut (..), from, to)
+import PlutusLedgerApi.V2.Contexts (getContinuingOutputs)
+import PlutusTx
+import PlutusTx.Bool
+import PlutusTx.Builtins
+import PlutusTx.Lift
+import PlutusTx.Maybe
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.Show qualified as PlutusTx
+import Prelude qualified as Haskell
+
+-- BLOCK1
+data AuctionParams = AuctionParams
+  { apSeller  :: PubKeyHash,
+    apAsset   :: Value,
+    apMinBid  :: Integer,
+    apEndTime :: POSIXTime
+  }
+
+PlutusTx.makeLift ''AuctionParams
+
+data Bid = Bid
+  { bBidder :: PubKeyHash,
+    bAmount :: Integer
+  }
+
+PlutusTx.deriveShow ''Bid
+PlutusTx.unstableMakeIsData ''Bid
+
+instance PlutusTx.Eq Bid where
+  {-# INLINEABLE (==) #-}
+  bid == bid' =
+    bBidder bid PlutusTx.== bBidder bid'
```

Comment:
I'm not sure why you're using all these operations qualified - would it not make more sense in a context like this to just import them unqualified?

Reply:
I guess it depends a bit on the pedagogical aims here? This tutorial seems quite light on the details of how exactly the Haskell is turning into Plutus Core, what's going on with the plugin, etc. , instead focussing on the _logic_ of the application. And I think that logic is clearer without all the qualification, we can make that point more clearly in a tutorial focussed on that, perhaps?